### PR TITLE
Consolidate course card cell

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-12410
affects: Student
release note: none

### CODE REVIEW

File: Student/Student/Courses/CourseList/CourseListViewController.swift [hunk 1]
```diff
@@ -52,6 +52,10 @@ class CourseListViewController: UIViewController, CourseListViewProtocol {
         // Collection View Setup
         collectionView?.refreshControl = refreshControl
 
+        let id = String(describing: CourseCardCell.self)
+        let nib = UINib(nibName: id, bundle: Bundle(for: type(of: self)))
+        collectionView?.register(nib, forCellWithReuseIdentifier: id)
```
Comment (vanwagonet) [inline Student/Student/Courses/CourseList/CourseListViewController.swift]: do we not have an extension for registering that mirrors the dequeue? That'd be nice.

Comment (shichongrui) [reply Student/Student/Courses/CourseList/CourseListViewController.swift]: I'll take a look.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-12410](https://instructure.atlassian.net/browse/MBL-12410)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.88% | 93.88% | 0%
**Student** | 92.51% | 92.51% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.71% | 92.71% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Extensions/UICollectionViewExtensions.swift [hunk 2]
```diff
@@ -37,7 +37,7 @@ extension UICollectionView {
 
     /// Registers a reusable cell object.
     /// This assumes that the reuse identifier matches the type name.
-    func registerCell(_ cellClass: AnyClass, bundle: Bundle? = nil) {
+    public func registerCell(_ cellClass: AnyClass, bundle: Bundle? = nil) {
```
Comment (vanwagonet) [inline Core/Core/Extensions/UICollectionViewExtensions.swift]: I don't know why this needs a bundle passed in. Won't the bundle always be the one `cellClass` is defined in?

**Pull request merged**